zeebe/engine/src/test/java/io/camunda/zeebe/engine/state/ReplayStateTest.java

```diff
@@ -405,51 +405,42 @@ public void shouldRestoreState() {
                     .isGreaterThanOrEqualTo(finalRecord.getPosition()));
 
     final var processingState = engine.collectState();
+    final var position = engine.getLastProcessedPosition();
     engine.stop();
 
     // when
-    engine.start();
+    engine.start(StreamProcessorMode.REPLAY, false);
```

Comment:
Hi @remcowesterhoud or @korthout,

I changed the `ReplayStateTest` slightly in https://github.com/camunda/camunda/pull/32760/commits/070e0db829b2a93a8a6c2525e00d91145e3668f2: It now restarts the engine in `REPLAY`. Previously, it started the engine without parameter and waited until it transitioned into `PROCESSING`.

The problem I encountered is that our `UsageMetricsChecker` is scheduled immediately as the engine is recovered and writes command record and eventually the applier alters the internals state, which causes this tests to fail.

❓My question is: Do you think there is anything this test verifies by waiting for the engine to get into `PROCESSING`? Or is it fine to _just_ replay.

Reply:
Looking at the test we waited for `PROCESSING` as a trigger that we could do the assertion. Just replaying would do the job fine. 

**However** I do question why we'd favor this over putting th `UsageMetricsChecker` behind a feature flag 😄

Reply:
We were also thinking of that. However currently the checker initializes the first bucket. I guess we could also make the tracking of metrics depending on the feature flag.
Can customers also change feature flags?